airflow/dag_processing/processor.py

```diff
@@ -795,8 +817,17 @@ def _execute_task_callbacks(self, dagbag: DagBag | None, request: TaskCallbackRe
         if task:
             ti.refresh_from_task(task)
 
-        ti.handle_failure(error=request.msg, test_mode=self.UNIT_TEST_MODE, session=session)
-        self.log.info("Executed failure callback for %s in state %s", ti, ti.state)
+        if callback_type is TaskInstanceState.SUCCESS:
+            context = ti.get_template_context(session=session)
+            if callback_type is TaskInstanceState.SUCCESS:
```

Comment:
Why do we want to check `if callback_type is TaskInstanceState.SUCCESS:` again here? are we expecting `context = ti.get_template_context(session=session)` to change the state?

Reply:
Yes that is correct we can not check it again as the state is not changed